src/firmware/arm_firmware/armSketchRev3.ino

```diff
@@ -0,0 +1,763 @@
+/*
+Created By: Tate Kolton, Rowan Zawadzki
+Created On: December 21, 2021
+Description: Main code for testing 6-axis stepper motor driven arm functionality. Interfaces with windows form through serial communication. Intended for use with Teensy 4.1
+Warning: May run into errors if attempting to use with Arduino Mega, Uno, or Nano. Insufficient DRAM / Flash, and clock speed. 
+*/
+
+#include <AccelStepper.h>
+
+// motor pins
+#define STEP_INDEX 0
+#define DIR_INDEX 1
+#define a1_step 3
+#define a1_dir 4
+#define a2_step 5
+#define a2_dir 5
+#define a3_step 6
+#define a3_dir 7
+#define a4_step 8
+#define a4_dir 9
+#define a5_step 10
+#define a5_dir 11
+#define a6_step 12
+#define a6_dir 13
+
+// limit switch pins
+#define SW_INDEX 2
+#define a1_sw 2
+#define a2_sw 48
+#define a3_sw 47
+#define a4_sw 46
+#define a5_sw 45
+#define a6_sw 44
+
+#define PPR_INDEX 4
+long a1_PPR = 400;
+long a2_PPR = 400;
+long a3_PPR = 400;
+long a4_PPR = 400;
+long a5_PPR = 400;
+long a6_PPR = 400;
+
+// axis speeds and accelerations
+#define SPEED_INDEX 3
+#define a1_SPEED 4500
+#define a2_SPEED 2000
+#define a3_SPEED 2000
+#define a4_SPEED 2000
+#define a5_SPEED 2000
+#define a6_SPEED 2000
+
+#define ACC_INDEX 4
+#define a1_ACC 5000
+#define a2_ACC 2000
+#define a3_ACC 2000
+#define a4_ACC 2000
+#define a5_ACC 2000
+#define a6_ACC 2000
+
+// homing speeds and accelerations
+#define a1_home_SPEED 1000
+#define a1_home_ACC 2000
+#define a2_home_SPEED 1000
+#define a2_home_ACC 2000
+#define a3_home_SPEED 1000
+#define a3_home_ACC 2000
+#define a4_home_SPEED 1000
+#define a4_home_ACC 2000
+#define a5_home_SPEED 1000
+#define a5_home_ACC 2000
+#define a6_home_SPEED 1000
+#define a6_home_ACC 2000
+
+// gear reduction ratios
+#define REDUCTION_INDEX 5
+#define a1_RED  161
+#define a2_RED  161
+#define a3_RED  150
+#define a4_RED  64
+#define a5_RED  57
+#define a6_RED  57
+
+// maximum angles of travel relative to limit switch
+#define ANG_INDEX 6
+#define a1_MAX_ANG 180
+#define a2_MAX_ANG 120
+#define a3_MAX_ANG 120
+#define a4_MAX_ANG 120
+#define a5_MAX_ANG 120
+#define a6_MAX_ANG 120
+
+// general parameters
+#define ENC_PPR 512
+#define NUM_AXES 6
+#define NUM_PARAMS 7
+#define ON 0
+#define OFF 1
+#define SW_ON 0
+#define SW_OFF 1
+
+// stepper motor objects for AccelStepper library
+AccelStepper a1(1, a1_step, a1_dir);
+AccelStepper a2(1, a2_step, a2_dir);
+AccelStepper a3(1, a3_step, a3_dir);
+AccelStepper a4(1, a4_step, a4_dir);
+AccelStepper a5(1, a5_step, a5_dir);
+AccelStepper a6(1, a6_step, a6_dir);
+
+// variable declarations
+int i, count, axis, temp, flag, sign,int_ang;
+long angle;
+char axisC, garbage, mode, serialBit;
+String char_angle;
+long angles[] = {0, 0, 0, 0, 0, 0};
+long max_steps[] = {a1_RED*a1_MAX_ANG/360.0*a1_PPR, a2_RED*a2_MAX_ANG/360.0*a2_PPR, a3_RED*a3_MAX_ANG/360.0*a3_PPR, a3_RED*a3_MAX_ANG/360.0*a3_PPR, a3_RED*a3_MAX_ANG/360.0*a3_PPR, a3_RED*a3_MAX_ANG/360.0*a3_PPR};
```

Comment:
Just curious, how come the last three element of this array don't use the gear reduction ratios/angles for the 4th, 5th, and 6th motors?

Reply:
good catch. Lazy copy and paste